lib/suma/spec_helpers/service.rb

```diff
@@ -338,14 +338,44 @@ def failure_message
   #   expect(last_response).to have_session_cookie
   #
   RSpec::Matchers.define(:have_session_cookie) do
-    look_for = Suma::Service::SESSION_COOKIE + "="
-
     match do |response|
-      response["Set-Cookie"]&.include?(look_for)
+      cookie = response["Set-Cookie"]
+      if cookie.nil?
+        @msg = "expected response to have a Set-Cookie header"
+        break false
+      end
+      cookie_prefix = Suma::Service::SESSION_COOKIE + "="
+      unless cookie.start_with?(cookie_prefix)
+        @msg = "expected response Set-Cookie '#{cookie}' to start with #{cookie_prefix}"
+        break false
+      end
+      payload = Suma::Service.decode_cookie(cookie)
+      (@payload_keys || []).each do |k|
```

Comment:
Where is @payload_keys coming from?

Reply:
https://github.com/lithictech/suma/pull/9/files#diff-172a7fe13aba1eab9588bdb48e19b61d9e8fa168c82dbbc2b53fd26ea58b81bdR369